pkg/resources/cloudcontroller/anexia.go

```diff
@@ -58,22 +58,38 @@ func anexiaDeploymentCreator(data *resources.TemplateData) reconciling.NamedDepl
 				return nil, fmt.Errorf("failed to get credentials: %w", err)
 			}
 
-			deployment.Spec.Template.Spec.Volumes = getVolumes(data.IsKonnectivityEnabled())
+			deployment.Spec.Template.Spec.Volumes = append(getVolumes(data.IsKonnectivityEnabled()),
+				corev1.Volume{
+					Name: resources.CloudConfigConfigMapName,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: resources.CloudConfigConfigMapName,
+							},
+						},
+					},
+				})
 
 			deployment.Spec.Template.Spec.Containers = []corev1.Container{
 				{
 					Name:  ccmContainerName,
-					Image: data.ImageRegistry(resources.RegistryAnexia) + "/anexia/anx-cloud-controller-manager:0.1.0",
+					Image: data.ImageRegistry(resources.RegistryAnexia) + "/anexia/anx-cloud-controller-manager:1.3.0",
 					Command: []string{
 						"/app/ccm",
 						"--cloud-provider=anexia",
+						"--cloud-config=/etc/kubernetes/cloud/config",
+						fmt.Sprintf("--cluster-name=%s", data.Cluster().Spec.HumanReadableName),
```

Comment:
This should probably set to Kubernetes controller-manager instead, as it is done here for azure: https://github.com/kubermatic/kubermatic/blob/c5daecbb129173098df5ddf82eb3c94482855d21/pkg/resources/controllermanager/deployment.go#L259

is there a reason this isn't just set for every cluster? should I extract this question into an issue?

Reply:
This depends on the CCM implementation for the cloud provider, like azure has support for this but others don't necessarily support this flag.

Reply:
That option is included in [`k8s.io/cloud-provider/options.KubeCloudSharedOptions`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/options/kubecloudshared.go#L59), but CCMs aren't required to use `k8s.io/cloud-provider` so yeah, correct, depends on CCM.